app/goal_routes.py

```diff
@@ -0,0 +1,101 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from app.helper_functions import validate_model
+
+goal_bp = Blueprint("goal", __name__, url_prefix="/goals")
+    
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal.from_dict(request_body)
+        
+        db.session.add(new_goal)
+        db.session.commit()
+
+        message  = {
+            "goal": new_goal.to_dict()
+        }
+        return make_response(message, 201)
+    except KeyError as e:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+@goal_bp.route("", methods=["GET"])
+def get_goals():    
+    goals = Goal.query.all()
+    goals_response = [goal.to_dict() for goal in goals]
+    return jsonify(goals_response) 
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    message  = {
+        "goal": goal.to_dict()
+        }
+    return make_response(message, 200)
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    message  = {
+        "goal": goal.to_dict()
+        }
+    return make_response(message, 200)
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    
+    db.session.delete(goal)
+    db.session.commit()
+
+    message  = {
+            "details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+        }
+
+    return make_response(message, 200)
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def create_tasks_for_goal_assignment(goal_id):
+    request_body = request.get_json()
+    goal = validate_model(Goal, goal_id)
+    task_list = request_body.get("task_ids")
```

Comment:
I like that you used .get() to get the task_ids. Consider passing a default value that is an empty list, like: `task_list = request_body.get("task_ids", [])`

As is, if the key "task_ids" is not in request_body then task_list will reference None. Then when you try to loop over None, you'll run into another unhandled exception (Nonetype is not iterable or something like that)